runtime/src/bank.rs

```diff
@@ -202,6 +202,23 @@ impl HashAgeKind {
     }
 }
 
+// This struct is a collection of all of various parameters used for creating a child bank from
+// parent bank. All passed parameters must be via this config to be hashed and SPV-ed. Only the
+// exception is previous_bank.hash().
+//
+// This struct propagates all of cumulative runtime state of ledger, equivalent to replaying
+// all the transactions since genesis.
+// It is named as XxxConfig because of similarity to the role of GenesisConfig,
+// which creates the genesis bank;
+// Its serialized binary is hashed into the `slot_hash` for the slot of a frozen bank.
+// So, this struct will be hashed at every slot boundary, so this is preferred to be small
+#[derive(Serialize)]
+pub struct BankConfig { // or BankState // slot_config
```

Comment:
Is the idea then that `Bank` is no longer serialized into snapshots?  Just this new `BankConfig` struct?

Reply:
@mvines Yeah, ideally, it could be made so for ideal security separation. However, I don't think that the serialization target change (Bank->BankConfig) is the scope of this PR at the moment. I want to minimize the required work here because of timing pressure. So, I want to just reuse some already-existing snapshot code as is, which is secured enough (Like accounts db setup).